### PR TITLE
chore(ai): upgrade openAI sdk to latest version

### PR DESCRIPTION
**Pull request by JoshFerge**
- [x] Upgrade OpenAI SDK to 1.3.5, the latest version
- [x] Fix OpenAI API calls to be compatible with new SDK changes
- [x] Fix typings, remove from typing blocklist (typings super helpful for verifying OpenAI SDK calls are correct)

As part of the LLM observability effort here: 
https://github.com/getsentry/getsentry/issues/12171

The upgrade helps us as we can focus on an integration w/ the latest OpenAI SDK API moving forward, and use this call to dogfood.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,7 +46,11 @@ looponfailroots = ["src", "tests"]
 [tool.mypy]
 python_version = "3.8"
 mypy_path = ["fixtures/stubs-for-mypy"]
-plugins = ["pydantic.mypy", "mypy_django_plugin.main", "tools.mypy_helpers.plugin"]
+plugins = [
+    "pydantic.mypy",
+    "mypy_django_plugin.main",
+    "tools.mypy_helpers.plugin",
+]
```
Comment (asottile-sentry) [inline pyproject.toml:53]: unrelated change

Comment (JoshFerge) [reply pyproject.toml:53]: yeah idk why my formatter wants to do that...

File: pyproject.toml [hunk 2]
```diff
@@ -46,7 +46,7 @@ looponfailroots = ["src", "tests"]
 [tool.mypy]
 python_version = "3.8"
 mypy_path = ["fixtures/stubs-for-mypy"]
-plugins = ["pydantic.mypy", "mypy_django_plugin.main", "tools.mypy_helpers.plugin"]
+plugins = ["pydantic.mypy","mypy_django_plugin.main","tools.mypy_helpers.plugin"]
```
Comment (asottile-sentry) [inline pyproject.toml:49]: unrelated change again :)

Comment (JoshFerge) [reply pyproject.toml]: 🪓

**Review by jennmueng — APPROVED** [review]
